### PR TITLE
feat: redo docs and guides general layout and nav

### PR DESCRIPTION
**Pull request by tnolet**
## Affected Components
* [ ] Content & Marketing
* [ ] Pricing
* [ ] Test
* [x] Docs
* [ ] Learn
* [ ] Other

## Pre-Requisites
* [ ] Code is linted (`npm run lint`)

## Notes for the Reviewer

The goal of this PR is to make the docs more unified with the Guides section and elevate "reference" to a category always visible in the side nav. Another goal is to remove most CTA / product top nav menus for a more focussed docs experience.

design:
- adds new navbar to `/docs` and `/guides` to unify the experience
- adds "reference" section to side nav.
- move search to topnav

docs:
- updates to reference docs

## Screenshots

Before:

CleanShot 2023-03-29 at 20 07 11

After:

CleanShot 2023-03-29 at 20 06 59

### CODE REVIEW

**Comment by tnolet** [comment]
@pilimartinez can you have a look at this PR and somehow make the mobile menu on the docs page look normal? Maybe CSS guru @stefanjudis know how to fix that mess. 
CleanShot 2023-03-29 at 20 18 51

**Comment by stefanjudis** [comment]
@tnolet a `flex-wrap: wrap` might do the trick. :) 

image

File: site/layouts/partials/docs-menu.html [hunk 1]
```diff
@@ -42,7 +35,45 @@
...
+        <div class="docs-menu-reference-link">
+          <a href="{{.URL}}">
+            {{ .Name }}
+          </a>
+        </div>
+      </li>
+      {{end}}
+    </ul>
+
+    <!--    <ul>-->
```
Comment (stefanjudis) [inline site/layouts/partials/docs-menu.html:52]: Commented code? 🤔

**Comment by stefanjudis** [comment]
@tnolet I'm super into this! 💯 Only one question: currently, you're linking into the CLI sections with the reference links. Do you plan on keeping doing this? Or will references become it's on section?

image

**Review by stefanjudis — DISMISSED** [review]
Only a nit and a question. :) But the CSS needs some adjustments to we'll have to do another round anyways.

**Comment by tnolet** [comment]
> @tnolet I'm super into this! 💯 Only one question: currently, you're linking into the CLI sections with the reference links. Do you plan on keeping doing this? Or will references become it's on section?
> 
> image

not sure yet.

**Comment by tnolet** [comment]
> @tnolet a `flex-wrap: wrap` on the header container might do the trick. :) There's still some margin / padding but hey. 😅
> 
> image
> 
> image

adding `flex-wrap` destroys the full width layout. @stefanjudis @pilimartinez can one of you (with more CSS knowledge) have a look at this?

**Comment by stefanjudis** [comment]
@tnolet I had a bit of a look, and you're right. It's a little trickier because of the nav toggle. As this is not my priority right now, I can try to fix the nav beginning next week if that's not too late for you. 

Here's a preview of what I have so far: 

image

**Comment by tnolet** [comment]
> @tnolet I had a bit of a look, and you're right. It's a little trickier because of the nav toggle. As this is not my priority right now, I can try to fix the nav beginning next week if that's not too late for you.
> 
> Here's a preview of what I have so far:
> 
> image

good enough for me. Can you push this to this branch?

**Comment by stefanjudis** [comment]
It's entirely broken on desktop, though. :D I don't think there's a quick CSS fix for what you're trying to do here. Shall I still push?

**Comment by tnolet** [comment]
> It's entirely broken on desktop, though. :D I don't think there's a quick CSS fix for what you're trying to do here. Shall I still push?

@stefanjudis no, if it's broken on desktop we don't update it. Can you or @pilimartinez have a look at this next week?

**Comment by tnolet** [comment]
let's go with the slightly scruffy mobile menu and merge this so I can continue with further updates. @stefanjudis

**Comment by stefanjudis** [comment]
Sure. Than you'll have it beginning of next week. :)

**Review by stefanjudis — APPROVED** [review]
:shipit:

**Pull request merged**